packages/grafana-data/src/valueFormats/categories.ts

```diff
@@ -318,6 +319,7 @@ export const getCategories = (): ValueFormatCategory[] => [
       { name: 'duration (ms)', id: 'dtdurationms', fn: toDurationInMilliseconds },
       { name: 'duration (s)', id: 'dtdurations', fn: toDurationInSeconds },
       { name: 'duration (hh:mm:ss)', id: 'dthms', fn: toDurationInHoursMinutesSeconds },
+      { name: 'duration (dd d hh:mm:ss)', id: 'dtdhms', fn: toDurationInDaysHoursMinutesSeconds },
```

Comment:
```suggestion
      { name: 'duration (dd _d_ hh:mm:ss)', id: 'dtdhms', fn: toDurationInDaysHoursMinutesSeconds },
```

Sorry for being a bit slow but what does the third `d` stand for?

Reply:
In the name it's **d**uration **t**o **d**ay **h**our **m**in **s**ec = dtdhms, parallels  existing **d**uration **t**o **h**our **m**in **s**ec = dthms.

In the parentheses it means the character `d` because there's no styling available to distinguish it from the `dd` indicating the day value.  Could express it as `DD d HH:MM:SS` but that'd be inconsistent with all the others.

Reply:
ah, Ignore the comment above. It is d = day(s)

Reply:
@pabigot how do you want this to work? I would probably expect the format to be `(dd hh:mm:ss)` to be consistent with the other formats in that list.

If the duration is 5 days, 2 hours, 15 minutes and 20 seconds. Will we then write the following: `05 d 02:15:20`?

Reply:
Otherwise I would probably only have the following `(d hh:mm:ss)`

Reply:
That's a possibility.  There's some bikeshedding in https://github.com/grafana/grafana/issues/8781#issuecomment-432631725 where `dd:hh:mm:ss` is ruled out (very difficult to comprehend quickly).

I prefer the explicit separating `d`, but could tolerate `5 02:15:20`.  I do find it harder to read without the separator.

There should be no leading zeros for the days field as it can be any size, and the prefix element is eliminated if the value would be zero.

It seems the unit test should capture days less than 10, and days more than 99, to make that clear.

Reply:
Sorry for being so unclear 🙈 It is totally my fault.

What I meant was basically the description of the duration option. So in the drop down might be changed to `d hh:mm:ss` to have the same format as the other ones. 

```suggestion
      { name: 'duration (d hh:mm:ss)', id: 'dtdhms', fn: toDurationInDaysHoursMinutesSeconds },
```

I totally agree with you that the formatted  duration value still should have the `5 d 02:15:20` as you suggested. It is much more readable.

Reply:
The reason why I am suggesting only one `d` in the beginning of the format is because to indicate that we won't append a 0 if the number of days is less then 10.

Basically following the same convention that we have in momentjs: https://momentjs.com/docs/#/displaying/format/ where e.g. M (Month) is formatted as 1 and MM (Month) is formatted as 01.

Reply:
And the reason why I am suggesting to remove the `d` that represents the day from the format in the description is because we don't have those kind of things for the other duration formats. They ads `ago`, `weeks`, `days`, `hours` and other units to the output without it being presented in the options drop down.

Reply:
Thanks; that's all clear now.  I'll rework this later today.